text/0000-cargo-report-future-incompat.md

```diff
@@ -306,6 +306,19 @@ As noted above, we want to continue to suppress normal lint checks for
 upstream dependencies. Therefore, Cargo will continue to pass
 `--cap-lints=allow` for non-path upstream dependencies.
 
+At the same time, we want to minimize disruption to existing users of Rust.
+Therefore, the behavior of flags that directly interact with lints, like
+`-Dwarnings`, will remain unchanged by this RFC.
+
+For example, in our running example of `unwary`:
+  * running `cargo rustc -- -Dwarnings` will behave the same as today
+    (the lints in the downstream `brash` dependency will be capped
+    and their warnings hidden), and
+  * running `RUSTFLAGS=-Dwarnings` will also behave the same as today
+    (the `rustc` invocation on the downstream `brash` depedency
+    will be invoked with `-Dwarnings` and thus the build of the `brash`
+    dependency will fail).
```

Comment:
Thanks for attempting to explicitly clarify the (lack of) impact to `-Dwarnings`, but I'm left confused/concerned regarding the example details you give.  AFAIK, currently, cargo would pass `--cap-lints=allow` to the build of the `brash` dependency, making _it not fail_, even when `RUSTFLAGS=-Dwarnings` is set in the environment.  I can give references to CI builds with this in place (and warnings in dependencies) if that helps.

Reply:
Ugh, sorry: I was testing via a local workspace with both crates as local paths, and I think that (once again) burned me.

cargo's behavior is pretty frustrating in this scenario where all I'm trying to do is directly explore the current semantics.

Is there a way to force it to treat a local path dependency the same way that it would treat a remote dependency?

(But also: I am a little surprised that passing `-Dwarnings` to `rustc` via the RUSTFLAGS env var does not override the passing of `--cap-lints`... but what do I know...)

Reply:
NP and yes, similarly but in the other direction, I have been surprised by the local-path behavior changing when cargo sends `--cap-lints=allow`.  I don't know of a way around it.    I think if you just mention that currently, the combination of (env) `RUSTFLAGS=-Dwarnings` and (cargo provided) `--cap-lints=allow` does not _fail_ the build, in combination with you stated goal of no practical change (on which we agree), then my concern is satisfied. Thanks!